src/norms.jl

```diff
@@ -100,34 +100,58 @@ GenericSOCExpr{C,V} = GenericNormExpr{2,C,V}
 # Alias for ‖Ax‖₂ and AffExpr case
 const SOCExpr = GenericSOCExpr{Float64,Variable}
 
-# getvalue(n::GenericNormExpr) = n.coeff * getvalue(n.norm) + getvalue(n.aff)
-#
-# ##########################################################################
-# # GenericSOCConstraint
-# # Second-order cone constraint of form
-# # α||Ax-b||₂ + cᵀx + d ≤ 0
-# type GenericSOCConstraint{T<:GenericSOCExpr} <: AbstractConstraint
-#     normexpr::T
-#     function (::Type{GenericSOCConstraint{T}}){T}(normexpr::T)
-#         if normexpr.coeff < 0
-#             # The coefficient in front of the norm is negative, which
-#             # means we have `norm >= c`, which is not convex.
-#             error("Invalid second-order cone constraint $(normexpr) ≤ 0")
-#         end
-#         new{T}(normexpr)
-#     end
-# end
-#
-# # Alias for the AffExpr case
-# const SOCConstraint = GenericSOCConstraint{SOCExpr}
-#
-# """
-#     addconstraint(m::Model, c::SOCConstraint)
-#
-# Add a SOC constraint to `Model m`.
-# """
-# function addconstraint(m::Model, c::SOCConstraint)
-#     push!(m.socconstr,c)
-#     m.internalModelLoaded = false
-#     ConstraintRef{Model,SOCConstraint}(m,length(m.socconstr))
-# end
+getvalue(n::GenericNormExpr) = n.coeff * getvalue(n.norm) + getvalue(n.aff)
+
+##########################################################################
+# GenericSOCConstraint
+# Second-order cone constraint of form
+# α||Ax-b||₂ + cᵀx + d ≤ 0
```

Comment:
I would question if we should use this data structure anymore. To be more consistent with MOI, a SOC constraint should just be a vector of affine expressions in the same order as the cone. This new structure could then be generalized to represent any VectorAffineFunction-in-set constraint.